### PR TITLE
Feature/i18n support

### PR DESCRIPTION
**Pull request by guillermodoghel**
Adds language support for en-EN and es-ES
Adds currency support for USD and EUR
Sends locale and currency properties to the popup window
Fixes issue with lib `"@types/react": "^17.0.24"`, moving to `"@types/react": "17.0.34"`,

Screenshot 2023-02-14 at 12 34 11 PM

Screenshot 2023-02-14 at 12 29 25 PM

Screenshot 2023-02-14 at 12 26 38 PM
Screenshot 2023-02-14 at 12 26 34 PM

Screenshot 2023-02-14 at 12 26 57 PM
Screenshot 2023-02-14 at 12 26 46 PM

### CODE REVIEW

File: packages/ui/react-ui/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
         "test": "jest"
     },
     "dependencies": {
-        "@crossmint/client-sdk-base": "^0.2.1-alpha.0",
+        "@crossmint/client-sdk-base": "file:../../core/base",
```
Comment (pgarciaegido) [inline packages/ui/react-ui/package.json:35]: maybe we want to take this back to what it was?

Comment (guillermodoghel) [reply packages/ui/react-ui/package.json:35]: Definetly, my question here: how does this work  on this specific case? 
UI needs CORE changes that are also submitted on this PR

File: packages/core/base/src/utils/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        ACCEPTED: "¡Está listo para usar Crossmint!",
+        REJECTED: "Su solicitud fue rechazada"
+    }
+}
+
+const localeMap = {
+    "en-US": enUS,
+    "es-ES": esES,
+}
+export function t(wordingKey: string, locale: Locale) {
```
Comment (mPaella) [inline packages/core/base/src/utils/i18n.ts:39]: `wordingKey` being a string isnt great

Can we either make it a type literal, or object

File: packages/ui/react-ui/src/CrossmintPayButton.tsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ export const CrossmintPayButton: FC<CrossmintPayButtonReactProps> = ({
     preferredSigninMethod,
     dismissOverlayOnClick,
     prepay,
+    locale="en-US",
+    currency="USD",
```
Comment (mPaella) [inline packages/ui/react-ui/src/CrossmintPayButton.tsx:44]: nit: spacing between `=`

File: packages/ui/vanilla-ui/src/CrossmintPayButton.ts [hunk 4]
```diff
@@ -37,63 +37,71 @@ const propertyDefaults: CrossmintPayButtonLitProps = {
...
     dismissOverlayOnClick: false,
     prepay: false,
+    locale: 'en-US',
+    currency: 'USD'
 };
 
 @customElement("crossmint-pay-button")
 export class CrossmintPayButton extends LitElement {
-    @property({ type: String })
+    @property({type: String})
```
Comment (mPaella) [inline packages/ui/vanilla-ui/src/CrossmintPayButton.ts:46]: these style changes dont seem correct?

Comment (mPaella) [reply packages/ui/vanilla-ui/src/CrossmintPayButton.ts:46]: seems like other files too

File: packages/core/base/src/utils/i18n.ts (hunk 2 above)
Comment (guillermodoghel) [reply packages/core/base/src/utils/i18n.ts]: Wanted to keep the "object.path.format", cause its kind of a standard, so found a way to make it work!

https://javascript.plainenglish.io/advanced-typescript-type-level-nested-object-paths-7f3d8901f29a

**Review by pgarciaegido — APPROVED** [review]

**Pull request merged**